yt_dlp/extractor/rambler.py

```diff
@@ -0,0 +1,137 @@
+        r'https?://vp\.rambler\.ru/player(?:/[\d.]+)?/(?P<type>embed|player)\.html',
+    ]
+    _TESTS = [{
+        'url': 'https://auto.rambler.ru/roadaccidents/54816856-v-moskve-mashina-s-diplomaticheskimi-nomerami-vrezalas-v-pushku',
+        'info_dict': {
+            'id': 'record::356179de-2b91-4e01-9a2c-409dc2f3171d',
+            'ext': 'mp4',
+            'title': 'Автомобиль с дипномерами врезался в пушку',
+            'channel': 'ramblernews',
+            'display_id': '54816856-v-moskve-mashina-s-diplomaticheskimi-nomerami-vrezalas-v-pushku',
+            'duration': 26,
+            'thumbnail': r're:https?://.+\.(?:jpe?g|png)',
+        },
+    }, {
+        'url': 'https://news.rambler.ru/moscow_city/55584944-sinoptik-ilin-nastuplenie-meteorologicheskoy-zimy-v-moskve-perenositsya',
+        'info_dict': {
+            'id': 'record::7d02afbf-4f77-438e-b302-394ac617d5e4',
+            'ext': 'mp4',
+            'title': 'Метеорологическая зима наступит в Москве на следующей неделе',
+            'channel': 'ramblernews',
+            'display_id': '55584944-sinoptik-ilin-nastuplenie-meteorologicheskoy-zimy-v-moskve-perenositsya',
+            'duration': 77,
+            'thumbnail': r're:https?://.+\.(?:jpe?g|png)',
+        },
+    }, {
+        'url': 'https://vp.rambler.ru/player/1.151.0/player.html#id=record%3A%3Afdb0a65e-dac4-4e48-abcb-6ec09e02d88f&referrer=https%3A%2F%2Fnews.rambler.ru%2Fcommunity%2F55274681-v-peterburge-muzhchina-otbil-podrostka-u-sluzhby-bezopasnosti-metro%2F',
+        'info_dict': {
+            'id': 'record::fdb0a65e-dac4-4e48-abcb-6ec09e02d88f',
+            'ext': 'mp4',
+            'title': 'В Петербурге мужчина отбил подростка у службы безопасности метро',
+            'channel': 'ramblernews',
+            'duration': 65,
+            'thumbnail': r're:https?://.+\.(?:jpe?g|png)',
+        },
+    }, {
+        'url': 'https://vp.rambler.ru/player/embed.html?id=record::01a9982f-da18-4e95-b3a1-c5bcf6e112fb',
+        'info_dict': {
+            'id': 'record::01a9982f-da18-4e95-b3a1-c5bcf6e112fb',
+            'ext': 'mp4',
+            'title': 'Ученый раскрыл тайну зеленого свечения над Москвой',
+            'channel': 'ramblernews',
+            'duration': 16,
+            'thumbnail': r're:https?://.+\.(?:jpe?g|png)',
+        },
+    }]
+
+    def _real_extract(self, url):
+        display_id = None if 'vp.rambler.ru' in base_url(url) else self._match_id(url)
+
+        if display_id:
+            webpage = self._download_webpage(url, display_id)
+            preloaded_state = self._search_json(
+                r'window\.__PRELOADED_STATE__\s*=', webpage,
+                'preloaded state', display_id, transform_source=js_to_json)
+            rambler_id = traverse_obj(preloaded_state, (
+                'commonData', 'entries', 'entities', ..., 'video',
+                ('recordId', ('videoData', 'embed-id')), {str}, any, {require('rambler ID')}))
+            referrer = url
```

Comment:
Would it make sense to split this in to two extractors? We have like two pretty different code paths here. One could just care about the embed URLs, and the other could return a URL result to the embed URL.